### PR TITLE
Bazel aspects: allow sharing source jars and allow empty sourcejars

### PR DESCRIPTION
**Pull request by antonsviridov-src**
Closes GRAPH-948
Closes GRAPH-949

### Test plan

- Reproducers added to existing cases

### CODE REVIEW

File: scip-java/src/main/resources/scip-java/scip_java.bzl [hunk 1]
```diff
@@ -70,15 +70,15 @@ def _scip_java(target, ctx):
...
+        dir = ctx.actions.declare_directory(ctx.label.name + "/extracted_srcjar/" + source_jar.short_path)
         output_dir.append(dir)
     
         ctx.actions.run_shell(
             inputs = javac_action.inputs,
             outputs = [dir],
             mnemonic = "ExtractSourceJars",
             command = """
-                unzip {input_file} -d {output_dir}
+                [ "$(unzip -q -l {input_file} | wc -l)" -eq 0 ] || unzip {input_file} -d {output_dir}
```
Comment (antonsviridov-src) [inline scip-java/src/main/resources/scip-java/scip_java.bzl:81]: Inexplicably, unzip returns a 1 exit code if there are any _warnings_ (see Diagnostics on https://linux.die.net/man/1/unzip) with no way to turn it off.

So we just literally count the number of files..

**Comment by antonsviridov-src** [comment]
@JohnnyMorganz would you like to build this PR from source and build on your repos?
Alternatively if you know of any sizeable public bazel repo that uses srcjars, I could use it for testing instead.

**Review by kritzcreek — APPROVED** [review]
If it works, it works. As discussed yesterday :)

Ideally we'd wait with merging until getting positive feedback that it fixes the issue for the customer.

**Comment by JohnnyMorganz** [comment]
Thanks for checking in!

Looking good from our side, but I am seeing one issue with test targets. You can repro this by changing the `java_library` to a `java_test`:

```python
java_test(
    name = "testing",
    srcs = [
        "Foo.java",
        ":generated-srcjar",
    ],
    test_class = "Foo",
)
```

Running the indexing gives the following error:
```
$ /home/code/scip-java/out/bin/scip-java index --build-tool=bazel --bazel-scip-java-binary="/home/code/scip-java/out/bin/scip-java" --output=index-java.scip -- -- //test/
ERROR: One of the output paths 'bazel-out/k8-dbg--cd/bin/test/testing' (belonging to //test:testing) and 'bazel-out/k8-dbg--cd/bin/test/testing/extracted_srcjar/test/sources.srcjar' (belonging to //test:testing) is a prefix of the other. These actions cannot be simultaneously present; please rename one of the output files or build just one of them
ERROR: com.google.devtools.build.lib.actions.ArtifactPrefixConflictException: One of the output paths 'bazel-out/k8-dbg--cd/bin/test/testing' (belonging to //test:testing) and 'bazel-out/k8-dbg--cd/bin/test/testing/extracted_srcjar/test/sources.srcjar' (belonging to //test:testing) is a prefix of the other. These actions cannot be simultaneously present; please rename one of the output files or build just one of them
```

Sorry, unfortunately I do not know of any good public repo using srcjars to test with

**Comment by antonsviridov-src** [comment]
Thanks @JohnnyMorganz, I'll try to address in a separate PR.

Will rely on your reproductions then :)

**Pull request merged**